custom_components/control4/climate.py

```diff
@@ -0,0 +1,434 @@
+        ):
+            return HVACMode.OFF
+        return HVAC_MODES[self._extra_state_attributes["HVAC_MODE"]]
+
+    @property
+    def hvac_modes(self) -> list(str) | None:
+        """Returns current fan mode."""
+        active_modes = list()
+        c4modes = self._extra_state_attributes["HVAC_MODES_LIST"].split(",")
+        # _LOGGER.debug( "c4modes = %s", self.coordinator.data[self._idx]["HVAC_MODES_LIST"],)
+        for mode in c4modes:
+            # _LOGGER.debug( "a_c4mode = %s", mode,)
+            if mode in HVAC_MODES and HVAC_MODES[mode] not in active_modes:
+                active_modes.append(HVAC_MODES[mode])
+        if len(active_modes) == 0:
+            active_modes.append(HVACMode.OFF)
+        return active_modes
+
+    @property
+    def is_aux_heat(self) -> bool | None:
+        """Return true if aux heater is active.
+        Requires ClimateEntityFeature.AUX_HEAT.
+        """
+        return "Emergency" in self._extra_state_attributes["HVAC_MODE"]
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature currently set to be reached."""
+        if self.hvac_mode == HVACMode.HEAT:
+            if "HEAT_SETPOINT_F" in self._extra_state_attributes:
+                return self._extra_state_attributes["HEAT_SETPOINT_F"]
+        if self.hvac_mode == HVACMode.COOL:
+            if "COOL_SETPOINT_F" in self._extra_state_attributes:
+                return self._extra_state_attributes["COOL_SETPOINT_F"]
+        return None
+
+    @property
+    def target_temperature_high(self) -> float | None:
+        """Return the upper bound target temperature."""
+        if self.hvac_mode != HVACMode.HEAT_COOL:
+            return None
+        return self._extra_state_attributes["COOL_SETPOINT_F"]
+
+    @property
+    def target_temperature_low(self) -> float | None:
+        """Return the lower bound target temperature."""
+        if self.hvac_mode != HVACMode.HEAT_COOL:
+            return None
+        return self._extra_state_attributes["HEAT_SETPOINT_F"]
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        if self._extra_state_attributes["SCALE"] == "F":
+            return UnitOfTemperature.FAHRENHEIT
+        else:
+            return UnitOfTemperature.CELSIUS
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Flag supported features."""
+        features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+        )
+        if "Emergency" in self._extra_state_attributes["HVAC_MODES_LIST"]:
+            features = features | ClimateEntityFeature.AUX_HEAT
+        #    _LOGGER.debug( "Emergency Found: %s",
+        #        self._extra_state_attributes["HVAC_MODES_LIST"],
+        # )
```

Comment:
Could you remove commented out code here and elsewhere in this PR

Reply:
Yes